components/Switch.tsx

```diff
@@ -0,0 +1,401 @@
+    if (!amountInSmallestUnit) {
+      return;
+    }
+    const sendTransaction = async (
+      transaction: SubmittableExtrinsic<"promise", ISubmittableResult>,
+      transactionFee: string,
+    ) => {
+      setTransaction(transaction);
+      setFeeDisplay(transactionFee);
+    };
+
+    if (source.id === "assethub") {
+      if (destination.id === "assethub") {
+        return;
+      }
+      await submitAssetHubToParachainTransfer({
+        context,
+        beneficiary,
+        source,
+        destination,
+        amount: amountInSmallestUnit,
+        sourceAccount,
+        setError,
+        setBusyMessage,
+        sendTransaction,
+      });
+    } else {
+      const { pallet } = parachainConfigs[source.name];
+
+      submitParachainToAssetHubTransfer({
+        context,
+        beneficiary,
+        source,
+        amount: amountInSmallestUnit,
+        pallet,
+        sourceAccount,
+        setError,
+        setBusyMessage,
+        sendTransaction,
+      });
+    }
+  }, [context, beneficiary, source, destination, sourceAccount, token, amount]);
+
+  useEffect(() => {
+    handleTransaction();
+  }, [handleTransaction]);
+
+  const handleSufficientTokens = (result: boolean) => {
+    setSufficientTokenAvailable(result);
+  };
+  const handleTopUpCheck = (result: boolean) => {
+    setTopUpCheck(result);
+  };
+  const onSubmit = useCallback(async () => {
+    if (!transaction || !context) {
+      return;
+    }
+
+    // to do: better error information for the user.
+    try {
+      if (destination.id === "assethub" && !sufficientTokenAvailable) {
+        setError({
+          title: "Not Enough Sufficient tokens",
+          description: "Please follow the sufficient or existential deposit",
+          errors: [
+            {
+              kind: "toPolkadot",
+              code: toPolkadot.SendValidationCode.BeneficiaryAccountMissing,
+              message:
+                "Asset Hub requires that you hold specific tokens in order for an account to be active.",
+            },
+          ],
+        });
+        return;
+      } else if (!sufficientTokenAvailable) {
+        setError({
+          title: "Not Enough Sufficient tokens",
+          description:
+            "Please follow the sufficient or existential deposit, make sure the beneficiary has enough funds on the destination account.",
+          errors: [],
+        });
+        return;
+      }
+
+      const { signer, address } = polkadotAccounts?.find(
+        (val) => val.address === sourceAccount,
+      )!;
+      if (!signer) {
+        throw new Error("Signer is not available");
+      }
+      setBusyMessage("Waiting for transaction to be confirmed by wallet.");
+      await transaction.signAndSend(address, { signer }, (result) => {
+        setBusyMessage("Currently in flight");
+
+        if (result.isFinalized) {
+          setBusyMessage("");
+          toast.info("Transfer Successful", {
```

Comment:
you're not handling dispatch errors here

Reply:
better? https://github.com/Snowfork/snowbridge-app/commit/9cd1da27c669c03186fe6a87466519c94f98469a